GameEngine/GameBoard.cs

```diff
@@ -1,45 +1,41 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace GameEngine
 {
     public class GameBoard
     { 
         public List<BoardPositionType> Board { get; private set; }
-        public int OwlPosition { get; set; }
-
-
-        public GameBoard(int gameSizeMultiplier)
+        public List<int> OwlPositions { get; set; }
+        
+        public int NestPosition
+        {
+            get { return Board.Count - 1; }
+        }
+        
+        public GameBoard(int gameSizeMultiplier, int numberOfOwls = 6)
         {
             BuildBoard(gameSizeMultiplier);
-            OwlPosition = 0;
+            OwlPositions = Enumerable.Repeat(0, numberOfOwls).ToList();
         }
 
-        public void Move(CardType type)
+        public void Move(Play play)
         {
-            if(OwlPosition == Board.Count - 1)
-            {
-                throw new InvalidMoveException("Owl is already in the Nest");
-            }
-            if(type == CardType.Sun)
+            if (play.Card == CardType.Sun)
             {
                 throw new InvalidMoveException("Sun is not a valid movement type");
             }
 
-            var targetBoardPosition = Convert(type);
-            for (int i = OwlPosition + 1; i < Board.Count; i++)
-            {
-                if(Board[i] == targetBoardPosition || Board[i] == BoardPositionType.Nest)
-                {
-                    OwlPosition = i;
-                    break;
-                }
-            }
+            var owlToMove = FindOwl(play.Position);
+            var newPosition = FindDestinationPosition(play);
+
+            OwlPositions[owlToMove] = newPosition;
```

Comment:
💡 I'm liking the idea of that `Parliament` class more now that I see this indexing

Reply:
Added it. It uses a `HashSet` under the hood. It's working great!